dotnet/src/Azure.Iot.Operations.Connector/ConnectorTelemetryFactory.cs

```diff
@@ -0,0 +1,151 @@
+    /// <returns>A configured connector telemetry sender.</returns>
+    public Task<ConnectorTelemetrySender<T>> CreateSenderAsync<T>(
+        AssetIdentity asset,
+        DataSetDefinition dataSet,
+        string? topicPattern = null)
+        where T : class
+    {
+        // TODO: Create base SDK TelemetrySender<T>
+        // Related to: Integration with base SDK
+        // Integration point: Creates Azure.Iot.Operations.Protocol.Telemetry.TelemetrySender<T>
+        // Steps:
+        // 1. Determine topic pattern (use provided or default)
+        // 2. Create TelemetrySender<T> using base SDK factory methods or constructor
+        // 3. Pass _applicationContext, _mqttClient, _serializer, topic pattern
+        
+        // TODO: Apply topic pattern configuration
+        // Related to: MQTT topic configuration
+        // Should support:
+        // - Default topic pattern (e.g., "telemetry/{AssetName}/{DataSetName}")
+        // - Custom topic patterns with token replacement
+        // - Topic token mapping (AssetName, AssetUuid, DataSetName, DeviceName, etc.)
+        
+        // TODO: Implement topic token mapping
+        // Related to: Topic pattern token replacement
+        // Should replace tokens in topic pattern with actual values:
+        // - {AssetName} -> asset.Name
+        // - {AssetUuid} -> asset.Uuid
+        // - {DataSetName} -> dataSet.Name
+        // - {DeviceName} -> _connectorConfig.Device.Name
+        // - {EndpointName} -> _connectorConfig.Endpoint.Name
+        // etc.
+        
+        // TODO: Wrap base sender in ConnectorTelemetrySender
+        // Related to: Connector SDK wrapper layer
+        // Steps:
+        // 1. Create ConnectorTelemetrySender<T> instance
+        // 2. Pass base sender, metadata builder, asset, dataset, logger
+        // 3. Return wrapped sender
+        
+        // TODO: Add logging
+        // Related to: Observability
+        // Should log:
+        // - Creating telemetry sender for asset/dataset
+        // - Topic pattern being used
+        // - Any configuration warnings
+        
+        // TODO: Add error handling
+        // Related to: Error handling
+        // Should handle:
+        // - Invalid asset or dataset
+        // - Base sender creation failures
+        // - Topic pattern errors
+        
+        throw new NotImplementedException();
+    }
+    
+    /// <summary>
+    /// Creates connector telemetry senders for all datasets in an asset configuration.
+    /// </summary>
+    /// <typeparam name="T">The type of telemetry data to send.</typeparam>
+    /// <param name="assetConfig">Asset configuration with all its datasets.</param>
+    /// <returns>A dictionary mapping dataset names to their connector telemetry senders.</returns>
+    public async Task<IReadOnlyDictionary<string, ConnectorTelemetrySender<T>>> CreateSendersForAssetAsync<T>(
```

Comment:
Assets can be discovered separately, so I don't think we will need a bulk API for this. Just the single CreateSenderForAssetAsync should be sufficient